src/sketch/sketch.cpp

```diff
@@ -26,39 +26,58 @@ inline uint64_t doublehash(uint64_t hash1, uint64_t hash2) {
     return (hash1 + hash2) % SIGN_MOD;
 }
 
-// Seeds for small k-mers
-// Make sure they are symmetric so RC works
-const unsigned int small_k = 9;
-robin_hood::unordered_map<int, std::vector<std::vector<unsigned> > > kmer_seeds({
-    {6, {{1,1,0,1,1,0,1,1}}},
-    {7, {{1,1,1,0,1,0,1,1,1}}},
-    {8, {{1,1,0,1,1,1,1,0,1,1}}},
-    {9, {{1,0,1,1,1,1,1,1,1,0,1}}}
-});
+// Codon phased seeds ([1,0,0]_k-1, 1)
+KmerSeeds generate_seeds(std::vector<size_t> kmer_lengths,
+                         const bool codon_phased) {
+    std::sort(kmer_lengths.begin(), kmer_lengths.end());
+    if (kmer_lengths.front() < 3) {
+        throw std::runtime_error("Minimum k must be 3 or higher");
+    }
+
+    KmerSeeds seeds;
+    if (codon_phased) {
+        std::vector<unsigned> spaced = {1,0,0,1,0,0,1};
+        size_t curr_k = 3;
+        for (auto k : kmer_lengths) {
+            while (curr_k < k) {
+                spaced.push_back(0);
+                spaced.push_back(0);
+                spaced.push_back(1);
+                curr_k++;
+            }
+            seeds[k] = spaced;
+        }
+    } else {
+        for (auto k : kmer_lengths) {
+            std::vector<unsigned int> dense_seed(k, 1);
+            seeds[k] = std::move(dense_seed);
+        }
+    }
+
+    return(seeds);
+}
```

Comment:
Looks right to me - we should put a min_k limit in PopPUNK to match this too.

Reply:
Noted in https://github.com/johnlees/PopPUNK/issues/102